File: .NET/BinToBloom Backend/Models/Entities/WasteLog.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BinToBloom_Backend.Models.Entities
{
    [Table("WasteLogs")]
    public class WasteLog
    {
        [Key]
        [Column("log_id")]
        public int LogId { get; set; }

        [Required]
        [Column("pickup_id")]
        public int PickupId { get; set; }

        [Required]
        [Column("waste_type")]
        [StringLength(20)]
        [RegularExpression("^(FOOD|E-WASTE)$", ErrorMessage = "Waste type must be FOOD or E-WASTE")]
        public string WasteType { get; set; } = string.Empty;

        [Required]
        [Column("weight_kg")]
        [Range(0.1, 1000, ErrorMessage = "Weight must be between 0.1 and 1000 kg")]
        public decimal WeightKg { get; set; }

        [Column("collected_at")]
        public DateTime CollectedAt { get; set; } = DateTime.Now;

        [Column("photo_url")]
        [StringLength(500)]
        [Url(ErrorMessage = "Invalid URL format")]
        public string? PhotoUrl { get; set; }

        [Column("notes")]
        [StringLength(500)]
        public string? Notes { get; set; }

        [ForeignKey("PickupId")]
        public PickupRequest PickupRequest { get; set; } = null!;
    }
}

File: Springboot/bintobloom-backend/src/main/java/com/bintobloom/entity/WasteLog.java
package com.bintobloom.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;
import java.math.BigDecimal;

@Entity
@Table(name = "waste_logs")
@Data
@NoArgsConstructor
@AllArgsConstructor
@EntityListeners(AuditingEntityListener.class)
public class WasteLog {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "log_id")
    private Long logId;
    
    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "pickup_id", nullable = false)
    private PickupRequest pickupRequest;
    
    @NotNull(message = "Waste type is required")
    @Enumerated(EnumType.STRING)
    @Column(name = "waste_type", nullable = false, length = 100)
    private PickupRequest.WasteType wasteType;
    
    @NotNull(message = "Weight is required")
    @DecimalMin(value = "0.1", message = "Weight must be at least 0.1 kg")
    @DecimalMax(value = "1000.0", message = "Weight must not exceed 1000 kg")
    @Column(name = "weight_kg", nullable = false, precision = 8, scale = 2)
    private BigDecimal weightKg;
    
    @Column(name = "photo_url", length = 500)
    private String photoUrl;
    
    @Size(max = 500, message = "Notes must not exceed 500 characters")
    @Column(length = 500)
    private String notes;
    
    @CreatedDate
    @Column(name = "collected_at", nullable = false, updatable = false)
    private LocalDateTime collectedAt;
}